### ISSUE TITLE
Consider respecting `&loadplugins`

### ISSUE THREAD

**Issue by stellarhoof** [Shougo/dein.vim#244]
# Problems summary

When `--noplugins` is specified in the cmdline, `dein#add` sets `sourced=1` in the plugin's options dictionary, even though the plugin hasn't been sourced. This makes `dein#source` to skip the plugin when called later

## Expected

`sourced=0` in the plugin's options dictionary

## Environment Information (Required!)

 * dein.vim version(SHA1): `3271fba1a562062be91365428c2aeaf34b362918`

 * OS: `MacOS`

 * Vim/neovim version: `NVIM v0.2.3-24-gd9b3ebfed`

## Provide a minimal .vimrc with less than 50 lines (Required!)

```vim
set runtimepath+=~/.local/share/nvim/dein/repos/github.com/Shougo/dein.vim

if dein#load_state('~/.local/share/nvim/dein')
    call dein#begin('~/.local/share/nvim/dein')
    call dein#add('https://github.com/Shougo/dein.vim')
    call dein#add('https://github.com/Konfekt/FastFold')
    call dein#end()
    call dein#save_state()
endif
```

**Comment by stellarhoof** [comment]
A quick fix is to add
```vim
    if &loadplugins
        let plugin.sourced = 1
    endif
```
in `dein#util#_end()`

**Comment by Shougo** [comment]
OK.

**Comment by Shougo** [comment]
@stellarhoof The fix is not compatible with Vim 7.4.
